### PR TITLE
Block Library: Add the Comments Pagination block

### LINKED ISSUE #35007: Post Comments Pagination block
## Description

Displays a paginated navigation to next/previous set of comments, when applicable. This should be similar to the [Query Pagination block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination).

This block is part of the Post Comments Loop block, for more context please visit this [Tracking Issue](https://github.com/WordPress/gutenberg/issues/34994) or its [GitHub project](https://github.com/WordPress/gutenberg/projects/61).

## Potential Settings

- **Color settings:** This should be similar to the [Query Pagination block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination).
- **Pagination Arrow:** It allows users to select if they want none/Arrow/Chevron in the Next & Previous buttons. This should be similar to the [Query Pagination block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination).

## Requirements

- This block should only be available inside the [Post Comments Query Loop Block](https://github.com/WordPress/gutenberg/issues/34996).
- If there is just one page, we shouldn't show the pagination.

## References

This should be similar to the [Query Pagination block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination).

### PR DESCRIPTION
**Pull request by DAreRodz**
## Description

This block works as a container for other pagination blocks that render links to navigate through the list of comments. It basically works the same as the [Query Pagination block](packages/block-library/src/query-pagination), but for comments.

It has the same settings:
- changing layout and colors
- choosing different styles for next/prev arrows. 

Will close #35007

## How has this been tested?

1. Modify the Single Post template, adding a Comments Query Loop block
2. Add inside the Comments Pagination block
3. Check that the layout and color settings work (tested only in the editor)
4. Check that the arrow setting works as expected (tested only in the editor)

## Types of changes
- New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/comments-pagination/edit.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+/**
+ * Internal dependencies
+ */
+import { CommentsPaginationArrowControls } from './comments-pagination-arrow-controls';
+
+const TEMPLATE = [
+	[ 'core/query-pagination-previous' ],
+	[ 'core/query-pagination-numbers' ],
+	[ 'core/query-pagination-next' ],
```
Comment (DAreRodz) [inline packages/block-library/src/comments-pagination/edit.js:23]: These blocks are meant to be changed once the rest of the `comments-pagination` blocks are created.

File: packages/block-library/src/comments-pagination/edit.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+			].includes( innerBlock.name );
+		} );
+	}, [] );
+	const blockProps = useBlockProps();
+	const innerBlocksProps = useInnerBlocksProps( blockProps, {
+		template: TEMPLATE,
+		allowedBlocks: [
+			'core/query-pagination-previous',
+			'core/query-pagination-numbers',
+			'core/query-pagination-next',
```
Comment (DAreRodz) [inline packages/block-library/src/comments-pagination/edit.js:61]: Again, these blocks should be replaced with the `comments-pagination` ones.

File: packages/block-library/src/comments-pagination/block.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (cbravobernal) [inline packages/block-library/src/comments-pagination/block.json:1]: "$schema": "https://schemas.wp.org/trunk/block.json",

Comment (cbravobernal) [reply packages/block-library/src/comments-pagination/block.json:1]: We need to add the schema there.

**Review by cbravobernal — CHANGES_REQUESTED** [review]
Nice job! It seems that we have to include the schema in order to pass the tests.
Also, do we have to create the fixtures? In order to do that, we have to create the core__comments-pagination-block.html with the content that is created on the code editor. Then paste it and then run `npm fixtures:generate`

**Comment by DAreRodz** [comment]
Thanks for the review @c4rl0sbr4v0 :bow: You're right, I forgot to add the fixture for the new block. Fixed now. :+1:

**Comment by cbravobernal** [comment]
Next question I have is. Should we merge it before having the other ones (core/comment-pagination-*? How would we approach this?

Leaving as is right now, it does not crash, right?

**Comment by gziolo** [comment]
> Next question I have is. Should we merge it before having the other ones (core/comment-pagination-*? How would we approach this?

Yes, we can also land this PR without inner blocks and address that in follow-up PRs. If we take that route we should rather hide this block from the inserter temporarily.

**Comment by aristath** [comment]
The PR needs to be rebased as there are now conflicts with `trunk` 👍

**Comment by cbravobernal** [comment]
> Yes, we can also land this PR without inner blocks and address that in follow-up PRs. If we take that route we should rather hide this block from the inserter temporarily.
> The PR needs to be rebased as there are now conflicts with trunk 👍

I updated the code with the requests and created a new PR, cherry-picking @DAreRodz commits. It is easier for me than making the updates on their repo. And also I included in this PR a Merge commit without wanting to.

#36872 closes this PR.

**Pull request closed without merging**